db/migrate/20150903162623_assign_tenant.rb

```diff
@@ -0,0 +1,53 @@
+class AssignTenant < ActiveRecord::Migration
+  class Tenant < ActiveRecord::Base
+    # seed and return the current root_tenant
+    def self.root_tenant
+      Tenant.where(:ancestry => nil).first || Tenant.create!(:use_config_for_attributes => true)
+    end
+  end
+
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+
+    belongs_to :tenant, :class_name => AssignTenant::Tenant
+  end
+
+  class MiqAeNamespace < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+
+    belongs_to :tenant, :class_name => AssignTenant::Tenant
+  end
+
+  class MiqGroup < ActiveRecord::Base
+    belongs_to :tenant, :class_name => AssignTenant::Tenant
+  end
+
+  class Provider < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+
+    belongs_to :tenant, :class_name => AssignTenant::Tenant
+  end
+
+  class TenantQuota < ActiveRecord::Base
+    belongs_to :tenant, :class_name => AssignTenant::Tenant
+  end
+
+  class Vm < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+
+    belongs_to :tenant, :class_name => AssignTenant::Tenant
+  end
+
+  def change
+    models = [ExtManagementSystem, MiqAeNamespace, MiqGroup,
+              Provider, TenantQuota, Vm]
+
+    # only create a root tenant if there are records in the db
+    return unless MiqGroup.exists?
+
+    root_tenant = Tenant.root_tenant
+    models.each do |model|
+      model.where(:tenant_id => nil).update_all(:tenant_id => root_tenant.id)
```

Comment:
You don't need any of the `belongs_to` in this migration.  This AR query will not use it.